### PR TITLE
Tes 737 trailing slash

### PR DESCRIPTION
**Pull request by Jessevanbekkum**
Fixed an issue that users got errors if they configured the xl testview server with a trailing slash

### CODE REVIEW

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java [hunk 1]
```diff
@@ -1,65 +1,93 @@
 package com.xebialabs.xlt.ci.server;
```
Comment (rklaren) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: Any particular technical reason to switch to junit ?

I'd have appreciated it in a different pull.

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java [hunk 2]
```diff
@@ -98,30 +126,97 @@ public void fillsTestSpecificationIdItems() {
...
 
         xlTestServer.uploadTestRun("testspecid", fp, "**/*.xml", null, metadata, logger);
 
-        RecordedRequest request = xltStub.takeRequest();
+        RecordedRequest request = xltestviewMock.takeRequest();
+        verifyUploadRequest(request);
+    }
+
+    @Test
+    public void shouldImportWithTrailingSlash() throws IOException, InterruptedException, MessagingException {
```
Comment (rklaren) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: This does not actually test much? Since the problem with the extra slash only occurs when talking to the real server that does not recognized the request with the extra slash in there?

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java [hunk 3]
```diff
@@ -98,30 +126,97 @@ public void fillsTestSpecificationIdItems() {
...
+        String TRAILING_SLASH = "/";
+
+        XLTestServerImpl slashedXlTestServer = new XLTestServerImpl(String.format("http://127.0.0.1:%d" + TRAILING_SLASH, xltestviewMock.getPort()), null, cred);
+
+        xltestviewMock.enqueue(new MockResponse()
+                .addHeader("Content-Type", "application/json; charset=utf-8")
+                .setBody("{ \"testRunId\": \"testrunid\" }"));
+        FilePath fp = new FilePath(new File(this.getClass().getResource("/demo_test_results").getPath()));
+
+        PrintStream logger = new PrintStream(System.out);
```
Comment (rklaren) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: Don't spam system.out.

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java (hunk 2 above)
Comment (rklaren) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: All in all the test checking the extra URL is very heavy for what it actually tests now. E.g. verifying that XLTestServerImpl.createUrl strips out extra '/'. Which is actually testing a JVM class.

Comment (Jessevanbekkum) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: I do not agree. We test the usage of the JVM class, not the JVM class itself. shouldImportWithTrailingSlash tests what happens if somebody types in a trailing slash, and verifies that the call is still what it should be, without a double slash. This is exactly the scenario that went wrong, and what we want to test

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java (hunk 1 above)
Comment (Jessevanbekkum) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: The reason was that I tried to get the @JenkinsRule to work, which is a JUnit rule that boots a jenkins instance and allows integration testing. It would be very useful, but it was difficult, and I ended up abandoning it. This was a relic that stayed in here. I agree it should have been done in a different PR

File: src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java (hunk 3 above)
Comment (Jessevanbekkum) [inline src/test/java/com/xebialabs/xlt/ci/server/XLTestServerImplTest.java]: Agree, copy paste of the old code.

**Comment by amolenaar** [comment]
Closed because it's to big.
